tealiumlibrary/src/main/java/com/tealium/core/Tealium.kt

```diff
@@ -316,6 +317,57 @@ class Tealium @JvmOverloads constructor(val key: String, val config: TealiumConf
      */
     @Suppress("unused")
     fun killTraceVisitorSession() {
-        activityObserverListener.killTraceVisitorSession()
+        deepLinkHandler.killTraceVisitorSession()
+    }
+
+    /**
+     * Migrates persistent data from the Tealium Android (Java) library if present
+     * */
+    private fun migratePersistentData() {
+        val hashCode = (config.accountName + '.' +
+                    config.profileName + '.' +
+                    config.environment.environment).hashCode()
+        val legacySharedPreferences = config.application.getSharedPreferences("tealium.datasources.${Integer.toHexString(hashCode)}", 0)
+        if (legacySharedPreferences.all.isEmpty()) {
+            return
+        }
+        legacySharedPreferences.all.forEach { item ->
+            val key = item.key
+            val value = item.value
+
+            (value as? String)?.let {
+                dataLayer.putString(key, it, Expiry.FOREVER)
+            }
+            (value as? Boolean)?.let {
+                dataLayer.putBoolean(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Double)?.let {
+                dataLayer.putDouble(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Int)?.let {
+                dataLayer.putInt(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Long)?.let {
+                dataLayer.putLong(key, it, Expiry.FOREVER)
+            }?:
+            if (value is Set<*>) {
+                value.firstOrNull()?.let {
+                    when(it) {
+                        is String -> {
+                            val stringArray: MutableList<String> = ArrayList<String>()
+                            value.forEach { anyValue ->
+                                anyValue?.let { stringValue ->
+                                    if (stringValue is String) {
+                                        stringArray.add(stringValue)
+                                    }
+                                }
+                            }
+                            dataLayer.putStringArray(key, stringArray.toTypedArray())
+                        }
+                    }
+                }
+            }
+        }
+        legacySharedPreferences.edit().clear().apply()
```

Comment:
just a thought about whether we want to make this configurable in some way? thinking for those who are testing a migration that may wish to have both running side-by-side - this line definitely makes it a one-way only, and if you want to go back for any reason then you cant.